modules/data/src/main/java/io/fluo/webindex/data/fluo/UriCountExport.java

```diff
@@ -15,41 +15,47 @@
 package io.fluo.webindex.data.fluo;
 
 import java.text.ParseException;
-import java.util.ArrayList;
-import java.util.Collection;
-
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.fluo.api.data.Bytes;
+import io.fluo.api.data.Column;
+import io.fluo.api.data.RowColumn;
+import io.fluo.recipes.accumulo.export.DifferenceExport;
 import io.fluo.webindex.core.Constants;
 import io.fluo.webindex.core.DataUtil;
 import io.fluo.webindex.data.fluo.UriMap.UriInfo;
-import io.fluo.webindex.data.recipes.Transmutable;
 import io.fluo.webindex.data.spark.IndexUtil;
+import io.fluo.webindex.data.util.FluoConstants;
 import io.fluo.webindex.data.util.LinkUtil;
-import org.apache.accumulo.core.data.Mutation;
 import org.slf4j.LoggerFactory;
 
-public class UriCountExport implements Transmutable<String> {
-  public UriInfo prevCount = UriInfo.EMPTY;
-  public UriInfo newCount = UriInfo.EMPTY;
+public class UriCountExport extends DifferenceExport<String, UriInfo> {
 
-  public UriCountExport() {}
+  public UriCountExport() {
+    super(UriInfo.EMPTY, UriInfo.EMPTY);
+  }
 
-  public UriCountExport(UriInfo prevCount, UriInfo newCount) {
-    this.prevCount = prevCount;
-    this.newCount = newCount;
+  public UriCountExport(UriInfo oldCount, UriInfo newCount) {
+    super(oldCount, newCount);
   }
 
   @Override
-  public Collection<Mutation> toMutations(String uri, long seq) {
-    ArrayList<Mutation> mutations = new ArrayList<>(4);
-
-    createTotalUpdates(mutations, uri, seq, prevCount, newCount);
-    Mutation m = createDomainUpdate(uri, seq, prevCount, newCount);
-    if (m != null) {
-      mutations.add(m);
+  protected Map<RowColumn, Bytes> generateData(String uri, UriInfo val) {
+    if (val == null || val.equals(UriInfo.EMPTY)) {
+      return Collections.emptyMap();
     }
-    mutations.add(createPageUpdate(uri, seq, newCount));
-
-    return mutations;
+    Map<RowColumn, Bytes> rcMap = new HashMap<>();
+    Bytes linksTo = Bytes.of("" + val.linksTo);
+    rcMap.put(new RowColumn(createTotalRow(uri, val.linksTo), Column.EMPTY), linksTo);
+    String domainRow = getDomainRow(uri);
+    if (domainRow != null) {
+      String cq = String.format("%s:%s", IndexUtil.revEncodeLong(val.linksTo), uri);
```

Comment:
I have found that String.format can be really slow.  I suspect that string concatenation would be much faster in this case.  I also suspect the code would be shorter.